client/lib/purchases/index.ts

```diff
@@ -41,51 +41,166 @@ import type { ReactChild } from 'react';
 
 const debug = debugFactory( 'calypso:purchases' );
 
-export interface SiteWithPurchases {
+interface PurchaseWithStatus extends Purchase {
+	currentPurchaseStatus: string;
+}
```

Comment:
Why not simply update the `Purchase` type defined in `client/lib/purchases/types.ts` instead of creating a new type?